File: JavaScript/002_Четные_числа_Фибоначчи.js
// console.log('Четные числа Фибоначчи')
var _ = require('lodash');
const { performance } = require('perf_hooks');

function fib(n) {
  x = 0
  y = 1
  for (let i=0; i<n; i++){
    tmp =x
    x = y
    y += tmp
  }
  return y
}

let start = performance.now();

lsFib = (_.range(100).map(fib));

let total = 0;
for (let el of lsFib){
  if (el%2 == 0) {
    total += el
  }
}

console.log(total)
console.log((performance.now() - start)/1000)

File: C++/002_Четные_числа_Фибоначчи.cpp
#include<iostream>  
#include <time.h>
#include <math.h>  

using namespace std; 
  
double fib(int n) {
  double x = 0;
  double y = 1;
  double tmp = 0;
  for (int i=0; i<n; i++) {
    tmp = x;
    x = y;
    y += tmp;
  }
  return y;
}


int main() 
{ 
  clock_t start, end;
  start = clock();

  int len = 1000;
  int ls[len];
  for (int i=0; i<len; i++) {
    ls[i] = i;
  }
  long double total = 0;
  long double res;
  for (int i=0; i<len; i++) {
    res = fib(i);
    if (!fmod(res, 2)) {
      total += res;
      /* cout<< i << " " << res << endl; */ 
    }
  }
  end = clock();
  cout << total << endl;
  cout << (end - start) / ((double) CLOCKS_PER_SEC) ;


  return 0; 
} 
